### PR TITLE
Rework hostname handling

### PR DESCRIPTION
**Pull request by kradalby**
This PR builds on the work of @bravechamp, and implements the suggested changes from the comments.

It replaces #498 

Changes:
- ~Machines can have nicknames~ Machines now has a separate name used for DNS (settable by users)
- Added "friendly name" on `nodes list` output
- New command `nodes rename` to rename nodes

- [x] added integration tests
- [x] updated documentation if needed
- [x] updated CHANGELOG.md
- [x] GivenName needs to be unique (DB constraint)
- [x] GivenName sanitise should have a tiny hash when registering, to avoid collisions)

### CODE REVIEW

File: grpcv1.go [hunk 1]
```diff
@@ -222,6 +222,28 @@ func (api headscaleV1APIServer) ExpireMachine(
...
+
+	api.h.RenameMachine(
+		machine,
+		request.GetNewName(),
+	)
+
+	log.Trace().
+		Str("machine", machine.Name).
+		Time("expiry", *machine.Expiry).
+		Msg("machine expired")
```
Comment (restanrm) [inline grpcv1.go:245]: It seems to be wrong log message here

File: machine.go [hunk 2]
```diff
@@ -367,6 +368,15 @@ func (h *Headscale) ExpireMachine(machine *Machine) {
 	h.db.Save(machine)
 }
 
+// RenameMachine takes a Machine struct and sets the expire field to now.
```
Comment (restanrm) [inline machine.go]: Wrong comment here

**Review by restanrm — COMMENTED** [review]
it's a good PR in my opinion

**Comment by kradalby** [comment]
Please take another look @restanrm @juanfont

**Review by juanfont — APPROVED** [review]
This is is a go!

**Comment by restanrm** [comment]
Ok for me too !

File: CHANGELOG.md [hunk 3]
```diff
@@ -17,6 +17,8 @@
 - Add the ability to rename a nodes name [#560](https://github.com/juanfont/headscale/pull/560)
   - Node DNS names are now unique, a random suffix will be added when a node joins
   - This change contains database changes, remember to **backup** your database before upgrading
+- Add option to enable/disable logtail (Tailscale's logging infrastructure) [#596](https://github.com/juanfont/headscale/pull/596)
```
Comment (juanfont) [inline CHANGELOG.md:20]: 👀

**Review by juanfont — APPROVED** [review]

**Comment by nikp123** [comment]
Not gonna lie, even though the unique hostname naming thing is correct, this kinda breaks my setup. Since I relied on the ``device_name.user.tailscale`` domain structure.

**Comment by e-zk** [comment]
Same here @nikp123 breaks a few things for me. 
I'm planning to start my tailnet from scratch and re-do some of my personal infra so this gives me a push to do that

**Comment by nikp123** [comment]
> Same here @nikp123 breaks a few things for me. I'm planning to start my tailnet from scratch and re-do some of my personal infra so this gives me a push to do that personally

Thankfully I found a workaround, which is to edit the sql entry for the node and change the given_name value.

**Comment by kradalby** [comment]
A more permanent workaround to have the username in the url is to generate the records and add it to a file:
https://github.com/juanfont/headscale/blob/d1dbe4ece9ad0e5ef8c12ae1c0f786d72fc43eab/config-example.yaml#L316

Then you can just have a script looking at machines + users doing it for you.

**Pull request merged**